### ISSUE TITLE
gitbreaker prevents the node.js process from exiting

### ISSUE THREAD

**Issue by iulo** [jdalrymple/gitbeaker#3493]
**Description**

- Node.js version: try both v20.9.0 and v18.19.0
- Gitbeaker version: 39.26.2
- Gitbeaker release (cli, node, browser, core, requester-utils): node
- OS & version: macos 14.2

when use gitbreaker for request, the nodejs process cannot exit immediate after everything done.

**Steps to reproduce**
a simple demo
```js
import { Gitlab } from "@gitbeaker/rest";

const start = Date.now();

process.on("exit", (code) => {
  console.log(`About to exit with code: ${code}`);
  console.log(`cost time: ${Date.now() - start}ms`);
});

const api = new Gitlab({
  token: "",
});

async function run() {
  try {
    const res = await api.RepositoryFiles.showRaw(
      "gitlab-org/gitlab",
      "README.md",
      "master",
    );

    console.log(`fetch success, summary: ${res.slice(0, 8)}...`);
  } catch (e) {
    console.error(e);
  }
}

run();

```
you can run it on codesandbox

https://codesandbox.io/p/devbox/test-gitbreaker-hvrrzy?embed=1&file=%2Ftest.js&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clqc3z8vp00063b6i0zwtktpl%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clqc3z8vo00023b6iyo6wdgr4%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clqc3z8vo00043b6iviuxkq46%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clqc3z8vo00053b6i1v2l71jz%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B67.32990686196872%252C32.67009313803128%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clqc3z8vo00023b6iyo6wdgr4%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqc3z8vo00013b6ionba5uqz%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clqc3z8vo00023b6iyo6wdgr4%2522%252C%2522activeTabId%2522%253A%2522clqc3z8vo00013b6ionba5uqz%2522%257D%252C%2522clqc3z8vo00053b6i1v2l71jz%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clqc3z8vo00053b6i1v2l71jz%2522%257D%252C%2522clqc3z8vo00043b6iviuxkq46%2522%253A%257B%2522id%2522%253A%2522clqc3z8vo00043b6iviuxkq46%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clqc3z8vo00033b6ilywwlx98%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%252C%257B%2522id%2522%253A%2522clqc40gff006k3b6i2htrkdxg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clqc40gkj03ysefg6dnnfazsm%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clqc40gff006k3b6i2htrkdxg%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D

**Expected behaviour**
dont prevent process termination.

**Actual behaviour**
process will exit after a period of time.

**Possible fixes**
tried `why-is-node-running`, it seems to be caused by rateLimit, got log like this
```
...


# Timeout
node:internal/async_hooks:202                                              
node:internal/async_hooks:505                                              
node:internal/timers:160                                                   
node:internal/timers:194                                                   
/workspaces/workspace/node_modules/async-sema/lib/index.js:160              - setTimeout(() => sema.release(), delay);
file:///workspaces/workspace/node_modules/@gitbeaker/rest/dist/index.mjs:79
```

**Checklist**
- [x] I have checked that this is not a duplicate issue.
- [x] I have read the documentation.

**Comment by iulo** [comment]
OK, I found the reason...

gitbreaker got a timeunit = 60000ms , and the RateLimit s implemented using setTimeout, so the minimum exit time is also 60s. 

Maybe the timeunit should reduce the timeunit, like 1s ?

**Comment by jdalrymple** [comment]
Slight issue, the reason for the 60s, is because thats the unit that the Gitlab API uses for rate limits (ie. 5 requests per minute)

I should defintely handle hanging timeouts though. Ill add that modification!

**Comment by jdalrymple** [comment]
Working on this. Found a couple ways to accomplish it, but none that seemed solid enough. Trying out another idea atm, ill follow up soon!

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`39.30.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/39.30.0) :rocket:
